File: java/ride.java
/* Use the slash-star style comments or the system won't see your
   identification information */
/*
ID: yeahyeet
LANG: JAVA
TASK: ride
*/

import java.io.*;
import java.util.*;

class ride {
    public static void main(String[] args) throws IOException {
        BufferedReader f = new BufferedReader(new FileReader("ride.in"));
        PrintWriter out = new PrintWriter(new BufferedWriter(new FileWriter("ride.out")));
        String line1 = f.readLine();
        String line2 = f.readLine();
        StringTokenizer st1 = new StringTokenizer(line1);
        StringTokenizer st2 = new StringTokenizer(line2);
        String s1 = st1.nextToken();
        String s2 = st2.nextToken();
        int n1 = getVal(s1);
        int n2 = getVal(s2);
        if((n1 % 47) == (n2 % 47))
            out.println("GO");
        else
            out.println("STAY");
        out.close();                               
    }

    static int getVal(String s) {
        char[] c = s.toCharArray();
        int sum = 1;
        for (char ch : c) {
            int pos = (int)ch -(int)('A')+1;
            sum *= pos;
        }
        return sum;
    }
}

File: training/section1.2/ride.cpp
/* Use the slash-star style comments or the system won't see your
   identification information */
/*
ID: yeahyeet
TASK: ride
LANG: C++                 
*/
/* LANG can be C++11 or C++14 for those more recent releases */
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

int main()
{
    ofstream fout("ride.out");
    ifstream fin("ride.in");
    string a, b;
    fin >> a >> b;
    int val1 = 1, val2 = 1;
    for (int i = 0; i < a.size(); ++i)
        val1 *= a[i] - 'A' + 1;
    for (int i = 0; i < b.size(); ++i)
        val2 *= b[i] - 'A' + 1;
    fout << (val1 % 47 == val2 % 47 ? "GO" : "STAY") << endl;
    return 0;
}
